fal_scripts/fal_github_insights.py

```diff
@@ -0,0 +1,38 @@
+import os
+import pycurl
+import json
+import certifi
+from io import BytesIO
+import pandas as pd
+
+ACCESS_TOKEN = os.environ['GITHUB_TOKEN']
+
+g_url = 'https://api.github.com/repos/fal-ai/fal/traffic/'
+header_list = [f'Authorization: token {ACCESS_TOKEN}']
+metrics = {'clones': '', 'views': '', 'popular/paths': '', 'popular/referrers': ''}
+
+for metric in metrics.keys():
+    buffer = BytesIO()
+    c = pycurl.Curl()
+    c.setopt(c.URL, g_url+metric)
+    c.setopt(c.HTTPHEADER, header_list)
+    c.setopt(c.WRITEDATA, buffer)
+    c.setopt(c.CAINFO, certifi.where())
+    c.perform()
+    c.close()
+    metrics[metric] = buffer.getvalue().decode('utf-8')
+    metrics[metric] = json.loads(metrics[metric])
+
+fal_views = pd.DataFrame.from_dict(metrics['views']['views'])
+fal_clones = pd.DataFrame.from_dict(metrics['clones']['clones'])
+fal_ppaths = pd.DataFrame.from_dict(metrics['popular/paths'])
+fal_preferrers = pd.DataFrame.from_dict(metrics['popular/referrers'])
+
+fal_views['timestamp'] = fal_views['timestamp'].transform(lambda x: pd.Timestamp(x))
+fal_clones['timestamp'] = fal_clones['timestamp'].transform(lambda x: pd.Timestamp(x))
+fal_ppaths = fal_ppaths.drop('title', 1)
+
+write_to_source(fal_views, 'fal_github_insights', 'fal_views')
```

Comment:
curious what happens if we run this multiple times? do we get duplicate records? can we build this such that there is some deduplication?

Reply:
Yes, the data gets duplicated. I tested this before and didn't really focus on writing something to prevent it because I didn't really have the time. From this, some tables also need to be just updated instead of appended to them. We may need the ability to read tables in `fal run --before`. I will start working on this from the 24th.

Reply:
Maybe this illustrates we need a possible UPSERT operation for write_to_source?